llvm/lib/Target/AMDGPU/SIInstrInfo.cpp

```diff
@@ -6079,6 +6079,17 @@ loadMBUFScalarOperandsFromVGPR(const SIInstrInfo &TII, MachineInstr &MI,
   unsigned MovExecOpc = ST.isWave32() ? AMDGPU::S_MOV_B32 : AMDGPU::S_MOV_B64;
   const auto *BoolXExecRC = TRI->getRegClass(AMDGPU::SReg_1_XEXECRegClassID);
 
+  // Save SCC. Waterfall Loop may overwrite SCC.
+  Register SaveSCCReg;
+  bool SCCNotDead = (MBB.computeRegisterLiveness(TRI, AMDGPU::SCC, MI, 30) !=
+                     MachineBasicBlock::LQR_Dead);
+  if (SCCNotDead) {
```

Comment:
Nits: call this `SCCLive`? Even better, rewrite so you don't need it:
```suggestion
  Register SaveSCCReg;
  if (MBB.computeRegisterLiveness(TRI, AMDGPU::SCC, MI, 30) !=
      MachineBasicBlock::LQR_Dead) {
```
Then below you can just test:
```
  // Restore SCC.
  if (SaveSCCReg) ...
```